app/views/govuk_publishing_components/components/_layout_footer.html.erb

```diff
@@ -26,24 +48,51 @@
+              list_classes << "govuk-footer__list--columns-#{section[:columns]}" if section[:columns] unless single_item_list
             %>
             <div class="<%= width_class %> govuk-!-display-none-print">
-              <h2 class="govuk-footer__heading govuk-heading-m"><%= item[:title] %></h2>
+              <h2 class="govuk-footer__heading govuk-heading-m"><%= section[:title] %></h2>
                 <ul class="<%= list_classes.join(' ') %>">
-                  <% item[:items].each do |item| %>
+                  <% section[:items].each_with_index do |item, index| %>
                     <% if item[:href] && item[:text] %>
                       <li class="govuk-footer__list-item">
                         <%
                           attributes = {
                             class: "govuk-footer__link",
                             data: {
-                              "track-category": "footerClicked"
+                              "track-category": "footerClicked",
+                              "ga4-link": {
+                                "event_name": "navigation",
+                                "type": "footer",
+                                "index": {
+                                  "index_link": (index + 1).to_s,
+                                  "index_section": (ga4_section_index + 1).to_s,
+                                  "index_section_count": ga4_index_section_count.to_s
+                                },
+                                  "index-total": ga4_index_total.to_s,
+                                  "section": section[:title]
+                              }
                             }
-                          }.merge(item.fetch(:attributes, {}))
+                          }
+
+                          # The below code merges the existing data attributes on a link with any set above.
+                          # If a link already contains any of the same data attributes,
+                          # that existing attribute will take priority over any defined above.
+                          # This is so that more complex values can be set with Ruby helpers,
+                          # whereas defaults and simple attributes can be set above.
+
+                          if item[:attributes]
+                            if item[:attributes][:data]
+                              # Merge the keys of both data attribute objects together
+                              attributes[:data] = attributes[:data].merge(item[:attributes].fetch(:data, {}))
+                              # Delete as there is a parent-level merge after this, and we don't want item[:attributes][:data] overwriting attributes[:data] which now contains the merged key/values.
+                              item[:attributes].delete(:data)
+                            end
+                          end
+                          attributes = attributes.merge(item.fetch(:attributes, {}))
```

Comment:
I wonder if there's a simpler option here? If we want to have default GA4 attributes but also allow some to be passed to be overridden, could we instead just check to see if they exist and if not, add them? Something like

```Ruby
<% if item[:href] && item[:text] %>
  <li class="govuk-footer__list-item">
    <%
      attributes = {
        class: "govuk-footer__link",
        data: {
          "track-category": "footerClicked",
        }
      }
      attributes = attributes.merge(item.fetch(:attributes, {}))

      unless attributes[:data][:ga4_link]
        attributes[:data][:ga4_link] = {
          "event_name": "navigation",
          "type": "footer",
          "index": {
            "index_link": (index + 1).to_s,
            "index_section": (ga4_section_index + 1).to_s,
            "index_section_count": ga4_index_section_count.to_s
          },
          "index-total": ga4_index_total.to_s,
          "section": section[:title]
        }
      end
    %>
  </li>
<% end %>
```

Reply:
Having thought about this, I guess you're thinking that we'd maybe pass only part of this, say if we wanted to override just one attribute. That's probably sensible, although what's the case for needing to override any of it? Is this functionality needed at all?

Reply:
@andysellick Thanks, I can change it to that. The reason I wrote it the previous way is because
I was trying to remove some of the weird behaviour. Your example is simpler to understand, but if someone goes in and tries to add a new data attribute to the `attributes` object in the template, it won't be reflected on any GOVUK footer link. This is because the merge line replaces the _whole_ [:data] object or **whole** [:class] string defined in the template if one already exists on the link already. It doesn't "merge" the values together, it merges keys together, and the keys are [:data] and [:class] so the whole data attributes or class gets overwritten if one already exists on the link, instead of actual values all getting merged together from both objects. Hope that makes sense!

Reply:
Ah okay. So maybe something like this would be helpful? (example taken from the component wrapper helper)

```Ruby
      def extend_object(option, object)
        @options[option] ||= {}
        object.each_key do |key|
          @options[option][key] =
            if @options[option].key?(key)
              "#{@options[option][key]} #{object[key]}"
            else
              object[key]
            end
        end
      end
```

But the main question I think we need to answer is do we need to worry about extending these objects, or can we just assume that our tracking will be in here and not passed in from outside the component?